playbooks/troubleshooting/troubleshooting_guide.adoc

```diff
@@ -0,0 +1,724 @@
+ +
+ +
+
+=== Registry not showing contents of NFS mount (persistent volume)
+
+You may run into a situation with Persistent Volumes where the node
+which is hosting the registry shows that the mount is there:
+
+----
+192.168.200.99:/storage/vms/ose_nfs/stratus on /var/lib/origin/openshift.local.volumes/pods/66295ab3-e624-11e5-952e-0800273943e4/volumes/kubernetes.io~nfs/docker-registry-storage type nfs4 (rw,relatime,vers=4.0,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=192.168.200.60,local_lock=none,addr=192.168.200.99)
+----
+
+but upon investigating the pod itself, there is no content showing in
+**/registry. **Checking the PV and PVC doesn't show any issues:
+
+----
+{
+  "apiVersion": "v1",
+  "kind": "PersistentVolume",
+  "metadata": { "name": "docker-registry-storage"
+  },
+  "spec": { "capacity": { "storage": "50Gi" }, "accessModes": [ "ReadWriteOnce" ], "nfs": { "path": "/storage/vms/ose_nfs/stratus/", "server": "192.168.200.99" }, "persistentVolumeReclaimPolicy": "Recycle"
+  }
+}
+----
+
+----
+[root@master00 ~]# oc get pv
+NAME                      LABELS    CAPACITY   ACCESSMODES   STATUS    CLAIM                          REASON    AGE
+docker-registry-storage   <none>    50Gi       RWO           Bound     default/docker-storage-claim             39m
+----
+
+----
+apiVersion: "v1"
+kind: "PersistentVolumeClaim"
+metadata:
+  name: "docker-storage-claim"
+spec:
+  accessModes:
+    - "ReadWriteOnce"
+  resources:
+    requests:
+      storage: "50Gi"
+----
+
+
+----
+[root@master00 ~]# oc get pvc
+NAME                   LABELS    STATUS    VOLUME                    CAPACITY   ACCESSMODES   AGE
+docker-storage-claim   <none>    Bound     docker-registry-storage   50Gi       RWO           38m
+----
+
+==== Solution
+
+This is most likely an error during the editing of the deployment
+config. When assigning the volume the *--name=* section refers to the
+name of the volume to be overwritten. The template for the router is
+mounting the *registry-storage* volume to **/registry**. Creating new
+claims and volumes thus, will not have the desired effect. You need to
+overwrite the default which is **emptydir** with the PV.
+
+*WRONG COMMAND:*
+
+----
+oc volume deploymentconfigs/docker-registry --add --name=docker-registry-storage -t pvc --claim-name=docker-storage-claim --overwrite
+----
+
+This is overwriting/creating a volume called docker-registry-storage
+which will not be able to be mounted in */registry*
+
+*CORRECT COMMAND:*
+
+----
+oc volume deploymentconfigs/docker-registry --add --name=registry-storage -t pvc --claim-name=docker-storage-claim --overwrite
+----
+
+=== Hosts Can No Longer Resolve Each During Anisble Install
+
+During the install, firewalld is disabled in favor of iptables. This
+transition may not open port 53 which is required for DNS lookups.
+
+==== Solution
+
+Use a DNS source external to the installer (at least during the installation
```

Comment:
@oybed how do we solve this problem in our labs? we install with local dns by default, but were able to get around this somehow.